runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -256,7 +256,8 @@ void registerTable(@Nonnull String namespace,
                 this.runtime.getObjectsView().TXBuild().type(TransactionType.WRITE_AFTER_WRITE).build().begin();
                 boolean protoFileChanged = tryUpdateTableSchemas(allDescriptors);
                 CorfuRecord<TableDescriptors, TableMetadata> oldRecord = this.registryTable.get(tableNameKey);
-                if (oldRecord == null || protoFileChanged) {
+                boolean flagChanged = federatedFlagChanged(oldRecord, metadataBuilder);
```

Comment:
instead of checking specific fields like federatedFlagChanged, can we check if any of the fields have changed?
at minimum if the key, value and metadata are still the same?

Reply:
Also can we validate this with a simple unit test?

Reply:
I think so right - the client should not be opening the same table in different modes anyway, but to be sure perhaps we can exclude updating the `TableRegsitry` if the `peristentDataPath` is changing.